03-transactions.md

```diff
@@ -151,13 +210,33 @@ The sending node can use the HTLC-timeout transaction to timeout the HTLC once t
 
 #### Received HTLC Outputs
 
-This output sends funds to either the remote node after the HTLC-timeout or using the revocation key, or to an HTLC-success transaction with a successful payment preimage. The output is a P2WSH, with a witness script:
+This output sends funds to either the remote node after the HTLC-timeout or using the revocation key, or to an HTLC-success transaction with a successful payment preimage. The output is a P2WSH, with a witness script (no `option_anchor_outputs`):
+
+    # To remote node with revocation key
+    OP_DUP OP_HASH160 <RIPEMD160(SHA256(revocationpubkey))> OP_EQUAL
+    OP_IF
+        OP_CHECKSIG
+    OP_ELSE
+        <remote_htlcpubkey> OP_SWAP OP_SIZE 32 OP_EQUAL
+        OP_IF
+            # To local node via HTLC-success transaction.
+            OP_HASH160 <RIPEMD160(payment_hash)> OP_EQUALVERIFY
+            2 OP_SWAP <local_htlcpubkey> 2 OP_CHECKMULTISIG
+        OP_ELSE
+            # To remote node after timeout.
+            OP_DROP <cltv_expiry> OP_CHECKLOCKTIMEVERIFY OP_DROP
+            OP_CHECKSIG
+        OP_ENDIF
+    OP_ENDIF
+
+Or, with `option_anchor_outputs`:
 
     # To remote node with revocation key
     OP_DUP OP_HASH160 <RIPEMD160(SHA256(revocationpubkey))> OP_EQUAL
     OP_IF
         OP_CHECKSIG
     OP_ELSE
+        1 OP_CHECKSEQUENCEVERIFY OP_DROP
```

Comment:
IMO this should be moved to reside _only_ within the else clause. Otherwise the offering party is unable to timeout the HTLC as soon as it has expired, introducing a dependnacy (which we worked to removed in the past) with this new csv value and the absolute HTLC timeout value.

Reply:
Suppose we'd move it into the (deepest) else clause, isn't the htlc success transaction potentially consuming the carve-out?

Reply:
this check can be moved after the inner `OP_ENDIF` and then we don't have to include `OP_DROP`

Reply:
Assuming this to be resolved. Check moved after inner `OP_ENDIF`